x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/EnrichPolicyRunner.java

```diff
@@ -232,19 +243,104 @@ private static void validateField(Map<?, ?> properties, String fieldName, boolea
+        } else if (EnrichPolicy.RANGE_TYPE.equals(policy.getType())) {
+            resolveRangeTypeEnrichMapping(policy, mappings, resultListener);
         } else if (EnrichPolicy.GEO_MATCH_TYPE.equals(policy.getType())) {
-            matchFieldMapping = (builder) -> builder.field("type", "geo_shape");
+            createEnrichMappingBuilder((builder) -> builder.field("type", "geo_shape"), resultListener);
         } else {
-            throw new ElasticsearchException("Unrecognized enrich policy type [{}]", policy.getType());
+            resultListener.onFailure(new ElasticsearchException("Unrecognized enrich policy type [{}]", policy.getType()));
+        }
+    }
+
+    private void resolveRangeTypeEnrichMapping(
+        EnrichPolicy policy,
+        List<Map<String, Object>> mappings,
+        ActionListener<XContentBuilder> resultListener
+    ) {
+        String matchFieldPath = "properties." + policy.getMatchField().replace(".", ".properties.");
+        List<Map<String, String>> matchFieldMappings = mappings.stream()
+            .map(map -> ObjectPath.<Map<String, String>>eval(matchFieldPath, map))
+            .filter(Objects::nonNull)
+            .collect(Collectors.toList());
+
+        Set<String> types = matchFieldMappings.stream().map(map -> map.get("type")).collect(Collectors.toSet());
+
+        if (types.isEmpty()) {
+            resultListener.onFailure(
+                new ElasticsearchException(
+                    "No mapping type found for match field '{}' - indices({})",
+                    policy.getMatchField(),
+                    Strings.collectionToCommaDelimitedString(policy.getIndices())
+                )
+            );
+        } else if (types.size() > 1) {
+            resultListener.onFailure(
+                new ElasticsearchException(
+                    "Multiple distinct mapping types for match field '{}' - indices({})  types({})",
+                    policy.getMatchField(),
+                    Strings.collectionToCommaDelimitedString(policy.getIndices()),
+                    Strings.collectionToCommaDelimitedString(types)
+                )
+            );
+        } else {
+            String type = types.iterator().next();
+            switch (type) {
+                case "integer_range":
+                case "float_range":
+                case "long_range":
+                case "double_range":
+                case "ip_range":
+                    createEnrichMappingBuilder((builder) -> builder.field("type", type).field("doc_values", false), resultListener);
+                    break;
+
+                // date_range types mappings allow for the format to be specified, should be preserved in the created index
+                case "date_range":
+                    Set<String> formatEntries = matchFieldMappings.stream().map(map -> map.get("format")).collect(Collectors.toSet());
+
+                    if (formatEntries.isEmpty()) {
```

Comment:
Same here (first valid cases and then error case)?

Reply:
addressed in https://github.com/elastic/elasticsearch/pull/76110/commits/114a452ca9ea4fd844788e53406c0b69da75415d